### ISSUE TITLE
[BUG] Cannot add new Service after fresh install

### ISSUE THREAD

**Issue by derelict** [bunkerity/bunkerweb#653]
### What happened?

Fresh autoconf based docker deployment. Can not add Service. Sometimes UI just hangs after creation ... or Service is not created at all.

### How to reproduce?

Fresh autoconf based docker deployment.

### Configuration file(s) (yaml or .env)

```YAML
version: "3.5"

services:
  bunkerweb:
    image: bunkerity/bunkerweb:1.5.2
    ports:
      - 80:8080
      - 443:8443
    labels:
      - "bunkerweb.INSTANCE"
    environment:
      - SERVER_NAME=
      - DATABASE_URI=mariadb+pymysql://bunkerweb:PWCHANGED@bw-db:3306/db
      - AUTOCONF_MODE=yes
      - MULTISITE=yes
      - API_WHITELIST_IP=127.0.0.0/8 10.20.30.0/24
    networks:
      - bw-universe
      - bw-services

  bw-autoconf:
    image: bunkerity/bunkerweb-autoconf:1.5.2
    depends_on:
      - bunkerweb
      - bw-docker
    environment:
      - DATABASE_URI=mariadb+pymysql://bunkerweb:PWCHANGED@bw-db:3306/db
      - AUTOCONF_MODE=yes
      - DOCKER_HOST=tcp://bw-docker:2375
    networks:
      - bw-universe
      - bw-docker

  bw-scheduler:
    image: bunkerity/bunkerweb-scheduler:1.5.2
    depends_on:
      - bunkerweb
      - bw-docker
    environment:
      - DATABASE_URI=mariadb+pymysql://bunkerweb:PWCHANGED@bw-db:3306/db
      - DOCKER_HOST=tcp://bw-docker:2375
      - AUTOCONF_MODE=yes
    networks:
      - bw-universe
      - bw-docker

  bw-docker:
    image: tecnativa/docker-socket-proxy:nightly
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - CONTAINERS=1
      - LOG_LEVEL=warning
    networks:
      - bw-docker

  bw-db:
    image: mariadb:10.10
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=yes
      - MYSQL_DATABASE=db
      - MYSQL_USER=bunkerweb
      - MYSQL_PASSWORD=PWCHANGED
    volumes:
      - bw-data:/var/lib/mysql
    networks:
      - bw-docker

  bw-ui:
    image: bunkerity/bunkerweb-ui:1.5.2
    networks:
      bw-docker:
      bw-universe:
        aliases:
          - bw-ui
    environment:
      - DATABASE_URI=mariadb+pymysql://bunkerweb:PWCHANGED@bw-db:3306/db
      - DOCKER_HOST=tcp://bw-docker:2375
      - AUTOCONF_MODE=yes
      - ADMIN_USERNAME=admin
      - ADMIN_PASSWORD=PWCHANGED
    labels:
      - "bunkerweb.SERVER_NAME=www.example.com"
      - "bunkerweb.USE_UI=yes"
      - "bunkerweb.USE_REVERSE_PROXY=yes"
      - "bunkerweb.REVERSE_PROXY_URL=/changeme"
      - "bunkerweb.REVERSE_PROXY_HOST=http://bw-ui:7000"
      - "bunkerweb.INTERCEPTED_ERROR_CODES=400 404 405 413 429 500 501 502 503 504"

volumes:
  bw-data:

networks:
  bw-universe:
    name: bw-universe
    ipam:
      driver: default
      config:
        - subnet: 10.20.30.0/24
  bw-services:
    name: bw-services
  bw-docker:
    name: bw-docker
```

### Relevant log output

```shell
[2023-09-22 16:39:56] - GENERATOR - ℹ️  - Variables : /tmp/cc7f3772-1cfd-4b34-8d46-53b3818ba3d4.env
[2023-09-22 16:39:56] - GENERATOR - ℹ️  - Computing config ...
[2023-09-22 16:39:56] - GENERATOR - ℹ️  - Database connection established
[2023-09-22 16:39:56] - GENERATOR - ℹ️  - Database not initialized, initializing ...
[2023-09-22 16:39:57] - GENERATOR - ❌ - Exception while initializing database : Traceback (most recent call last):
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1934, in _exec_single_context
    self.dialect.do_executemany(
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/default.py", line 918, in do_executemany
    cursor.executemany(statement, parameters)
sqlite3.IntegrityError: UNIQUE constraint failed: bw_plugins.id
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/share/bunkerweb/db/Database.py", line 474, in init_tables
    session.commit()
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 1923, in commit
    trans.commit(_to_root=True)
  File "<string>", line 2, in commit
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/state_changes.py", line 139, in _go
    ret_value = fn(self, *arg, **kw)
                ^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 1239, in commit
    self._prepare_impl()
  File "<string>", line 2, in _prepare_impl
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/state_changes.py", line 139, in _go
    ret_value = fn(self, *arg, **kw)
                ^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 1214, in _prepare_impl
    self.session.flush()
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 4179, in flush
    self._flush(objects)
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 4314, in _flush
    with util.safe_reraise():
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/util/langhelpers.py", line 147, in __exit__
    raise exc_value.with_traceback(exc_tb)
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/session.py", line 4275, in _flush
    flush_context.execute()
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/unitofwork.py", line 466, in execute
    rec.execute(self)
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/unitofwork.py", line 642, in execute
    util.preloaded.orm_persistence.save_obj(
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/persistence.py", line 93, in save_obj
    _emit_insert_statements(
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/orm/persistence.py", line 1043, in _emit_insert_statements
    result = connection.execute(
             ^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1412, in execute
    return meth(
           ^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/sql/elements.py", line 515, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1635, in _execute_clauseelement
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1844, in _execute_context
    return self._exec_single_context(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1984, in _exec_single_context
    self._handle_dbapi_exception(
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 2339, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/base.py", line 1934, in _exec_single_context
    self.dialect.do_executemany(
  File "/usr/share/bunkerweb/deps/python/sqlalchemy/engine/default.py", line 918, in do_executemany
    cursor.executemany(statement, parameters)
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: bw_plugins.id
[SQL: INSERT INTO bw_plugins (id, name, description, version, stream, external, method, data, checksum) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: [('antibot', 'Antibot', 'Bot detection by using a challenge.', '1.0', 'no', 0, 'manual', None, None), ('authbasic', 'Auth basic', 'Enforce login before accessing a resource or the whole site using HTTP basic auth method.', '1.0', 'no', 0, 'manual', None, None), ('badbehavior', 'Bad behavior', "Ban IP generating too much 'bad' HTTP status code in a period of time.", '1.0', 'yes', 0, 'manual', None, None), ('blacklist', 'Blacklist', 'Deny access based on internal and external IP/network/rDNS/ASN blacklists.', '1.0', 'partial', 0, 'manual', None, None), ('brotli', 'Brotli', 'Compress HTTP requests with the brotli algorithm.', '1.0', 'no', 0, 'manual', None, None), ('bunkernet', 'BunkerNet', 'Share threat data with other BunkerWeb instances via BunkerNet.', '1.0', 'yes', 0, 'manual', None, None), ('clientcache', 'Client cache', 'Manage caching for clients.', '1.0', 'no', 0, 'manual', None, None), ('cors', 'CORS', 'Cross-Origin Resource Sharing.', '1.0', 'no', 0, 'manual', None, None)  ... displaying 10 of 33 total bound parameter sets ...  ('ui', 'UI', 'Integrate easily the BunkerWeb UI.', '1.0', 'no', 0, 'manual', None, None), ('whitelist', 'Whitelist', 'Allow access based on internal and external IP/network/rDNS/ASN whitelists.', '1.0', 'partial', 0, 'manual', None, None)]]
(Background on this error at: https://sqlalche.me/e/20/gkpj)
Exception in thread Reloading instances:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.11/threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/share/bunkerweb/ui/main.py", line 256, in manage_bunkerweb
    operation, error = app.config["CONFIG"].new_service(args[0])
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/bunkerweb/ui/src/Config.py", line 214, in new_service
    self.__gen_conf(self.get_config(methods=False), services)
  File "/usr/share/bunkerweb/ui/src/Config.py", line 79, in __gen_conf
    raise Exception(f"Error from generator (return code = {proc.returncode})")
Exception: Error from generator (return code = 1)
10.10.10.10 - - [22/Sep/2023:16:39:58 +0000] "GET /check_reloading?next=/services&message=Creating+service+www.test.com HTTP/1.1" 200 24 "http://1.2.3.4:7000/loading?next=/services&message=Creating+service+www.test.com" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36"
10.10.10.10 - - [22/Sep/2023:16:40:00 +0000] "GET /check_reloading?next=/services&message=Creating+service+www.test.com HTTP/1.1" 200 24 "http://1.2.3.4:7000/loading?next=/services&message=Creating+service+www.test.com" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36"
10.10.10.10 - - [22/Sep/2023:16:40:02 +0000] "GET /check_reloading?next=/services&message=Creating+service+www.test.com HTTP/1.1" 200 24 "http://1.2.3.4:7000/loading?next=/services&message=Creating+service+www.test.com" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36"
10.10.10.10 - - [22/Sep/2023:16:40:04 +0000] "GET /check_reloading?next=/services&message=Creating+service+www.test.com HTTP/1.1" 200 24 "http://1.2.3.4:7000/loading?next=/services&message=Creating+service+www.test.com" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36"
10.10.10.10 - - [22/Sep/2023:16:40:06 +0000] "GET /check_reloading?next=/services&message=Creating+service+www.test.com HTTP/1.1" 200 24 "http://1.2.3.4:7000/loading?next=/services&message=Creating+service+www.test.com" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36"
10.10.10.10 - - [22/Sep/2023:16:40:08 +0000] "GET /check_reloading?next=/services&message=Creating+service+www.test.com HTTP/1.1" 200 24 "http://1.2.3.4:7000/loading?next=/services&message=Creating+service+www.test.com" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36"
```

### BunkerWeb version

1.5.2

### What integration are you using?

Autoconf

### Linux distribution (if applicable)

Ubuntu

### Removed private data

- [X] I have removed all private data from the configuration file and the logs

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by TheophileDiot** [comment]
Hi @derelict, thank you for opening this issue, this has been fixed with this commit https://github.com/bunkerity/bunkerweb/commit/33e200f652b1f2e8f704e43547e623507b0db124 🐛
And will be available in the next update ! 🚀

**Comment by derelict** [comment]
> Hi @derelict, thank you for opening this issue, this has been fixed with this commit [33e200f](https://github.com/bunkerity/bunkerweb/commit/33e200f652b1f2e8f704e43547e623507b0db124) 🐛 And will be available in the next update ! 🚀

Ok. Cool. Let me build a new Container based on the DEV Branch so i can give it a try before Release.

Thanks !

**Comment by TheophileDiot** [comment]
Any time 😁

**Comment by derelict** [comment]
i can confirm. It works now (DEV Branch). Thanks.

**Comment by TheophileDiot** [comment]
Happy to hear that 😁.
Let's keep this issue open while it isn't merge in the master branch in case someone else have this issue as well.

**Comment by andyoulovexy** [comment]
> Happy to hear that 😁. Let's keep this issue open while it isn't merged in the master branch in case someone else have this issue as well.

I test in docker tag dev,but clamav and virustotoal does't work.
This is my compose file:
--------
version: "3.5"

services:
  bunkerweb:
    image: bunkerity/bunkerweb:dev
    ports:
      - 80:8080
      - 443:8443
    labels:
      - "bunkerweb.INSTANCE"
    environment:
      - SERVER_NAME=xxxxxxx
      - MULTISITE=yes
      - USE_CLAMAV=yes
      - CLAMAV_HOST=clamav
      #- USE_VIRUSTOTAL=yes
      #- VIRUSTOTAL_API=http://virustotal-api:8000
      - CLAMAV_API=http://clamav-api:8000
      - VIRUSTOTAL_API_KEY=xxxxxx
      - DATABASE_URI=mariadb+pymysql://bunkerweb:xxxxxx@bw-db:3306/db # Remember to set a stronger password for the database
      - API_WHITELIST_IP=127.0.0.0/8 10.20.30.0/24
      - DISABLE_DEFAULT_SERVER=yes
      - USE_CLIENT_CACHE=yes
      - USE_GZIP=yes
      - xxxxx_USE_UI=yes
      - xxxxx_USE_REVERSE_PROXY=yes
      - xxxxx_REVERSE_PROXY_URL=/xxxxx
      - xxxxx_REVERSE_PROXY_HOST=http://bw-ui:7000
      - xxxxx_INTERCEPTED_ERROR_CODES=400 404 405 413 429 500 501 502 503 504
    networks:
      - bw-universe
      - bw-services
    logging:
      driver: syslog
      options:
        syslog-address: "udp://10.20.30.254:514"

  bw-scheduler:
    image: bunkerity/bunkerweb-scheduler:dev
    volumes:
      - ./bw-data:/data
    depends_on:
      - bunkerweb
      - bw-docker
    environment:
      - DATABASE_URI=mariadb+pymysql://bunkerweb:xxxxxx@bw-db:3306/db # Remember to set a stronger password for the database
      - DOCKER_HOST=tcp://bw-docker:2375
    networks:
      - bw-universe
      - bw-docker

  bw-docker:
    image: tecnativa/docker-socket-proxy:nightly
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - CONTAINERS=1
      - LOG_LEVEL=warning
    networks:
      - bw-docker

  bw-ui:
    image: bunkerity/bunkerweb-ui:dev
    depends_on:
      - bw-docker
    environment:
      - DATABASE_URI=mariadb+pymysql://bunkerweb:xxxxxx@bw-db:3306/db # Remember to set a stronger password for the database
      - DOCKER_HOST=tcp://bw-docker:2375
      - ADMIN_USERNAME=xxxxx
      - ADMIN_PASSWORD=xxxxxxe # Remember to set a stronger password for the Abc1234%Fsme user
    networks:
      - bw-universe
      - bw-docker

  crowdsec:
    image: crowdsecurity/crowdsec:v1.5.2
    volumes:
      - cs-data:/var/lib/crowdsec/data
      - ./acquis.yaml:/etc/crowdsec/acquis.yaml
      - bw-logs:/var/log:ro
      - /var/log/messages:/mnt/messages:ro
      - /var/log/secure:/mnt/secure:ro
    environment:
      - BOUNCER_KEY_bunkerweb=xxxxxxx
      - COLLECTIONS=crowdsecurity/nginx
    networks:
      - bw-universe

  syslog:
    image: balabit/syslog-ng:4.2.0
    volumes:
      - ./syslog-ng.conf:/etc/syslog-ng/syslog-ng.conf
      - bw-logs:/var/log
    networks:
      bw-universe:
        ipv4_address: 10.20.30.254


  bw-db:
    image: mariadb:10.10
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=yes
      - MYSQL_DATABASE=db
      - MYSQL_USER=xxxxxxx
      - MYSQL_PASSWORD=xxxxxx # Remember to set a stronger password for the database
    volumes:
      - bw-data:/var/lib/mysql
    networks:
      - bw-docker

  clamav:
    image: clamav/clamav:1.1
    volumes:
      - ./clamav-data:/var/lib/clamav
    networks:
      - bw-universe

  clamav-api:
    image: bunkerity/bunkerweb-clamav
    networks:
      - bw-universe
    environment:
      - CLAMAV_HOST=clamav
      - REDIS_HOST=redis

  virustotal-api:
    image: bunkerity/bunkerweb-virustotal
    environment:
      - API_KEY=xxxxxxxxx
      - REDIS_HOST=redis
    networks:
      - bw-universe

  redis:
    image: redis:7-alpine
    networks:
      - bw-universe

volumes:
  bw-data:
  cs-data:
  bw-logs:

networks:
  bw-universe:
    name: bw-universe
    ipam:
      driver: default
      config:
        - subnet: 10.20.30.0/24
  bw-services:
    name: bw-services
  bw-docker:
    name: bw-docker

**Comment by fl0ppy-d1sk** [comment]
Hello @derelict,

It should be fixed in the last 1.5.3 release. Don't hesitate to open a new issue if needed.
